# WIP: Find a way to make it possible to read in provedIt data and have an existing model make predictions?

In [7]:
%cd /Users/amarmesic/Documents/tudelft/thesis/DNANet

/Users/amarmesic/Documents/tudelft/thesis/DNANet


/Users/amarmesic/miniconda3/envs/dnanet/lib/python3.10/site-packages/IPython/core/magics/osm.py:417: UserWarning: This is now an optional IPython functionality, setting dhist requires you to install the `pickleshare` library.
  self.shell.db['dhist'] = compress_dhist(dhist)[-100:]


In [8]:
from DNAnet.data.data_models import Panel
from DNAnet.data.data_models.custom_hid_dataset import CustomHIDDataset
from DNAnet.data.parsing.file_categorization_strategy import ProvedItFileCategorizer
from DNAnet.data.validation.sample_validation_strategy import NFIValidationStrategy
from DNAnet.data.kit_compatibility.lane_standards import InternalSizeStandard


from typing import List
from pathlib import Path

# Create HID Image

In [9]:
gf_panel_path = "resources/data/ProvedIt/SGPanel_Globalfiler_Panel.xml"
hid_files_path = "/Users/amarmesic/Documents/tudelft/thesis/datasets/USE THIS - PROVEDIt_2-5-Person Profiles_3500 5sec_GF29cycles"
reference_genotype_path = "/Users/amarmesic/Documents/tudelft/thesis/DNANet/resources/data/ProvedIt/References"
file_categorizer = ProvedItFileCategorizer()
sample_validator = NFIValidationStrategy()

In [11]:
images = CustomHIDDataset(
    root_path=hid_files_path,
    panel_path=gf_panel_path,
    size_standard=InternalSizeStandard.GENESCAN_600_LIZ.value,
    file_categorization_strategy=file_categorizer,
    sample_validation_strategy=sample_validator,
)

In [13]:
len(images.sample_files), len(images)

(675, 518)

In [ ]:
from config_io import load_model

unet_model = load_model("config/models/unet.yaml")
unet_model.load("resources/model/current_best_unet/")
predictions = unet_model.predict_batch(images)

/Users/amarmesic/miniconda3/envs/dnanet/lib/python3.10/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


KeyboardInterrupt: 

In [ ]:
from DNAnet.evaluation.segmentation.allele_metrics import allele_f1_score, allele_precision, allele_recall

# current issue: we are also retrieving ladder files, and other files not conforming to the standard
print('allele u-net f1: ', allele_f1_score(images, predictions))
print('allele u-net precision: ', allele_precision(images, predictions))
print('allele u-net recall: ', allele_recall(images, predictions))

allele u-net f1:  0.8044332375535097
allele u-net precision:  0.8504649721016739
allele u-net recall:  0.7631286159323543


### 30 mixtures:
* allele u-net f1:  0.8227742115545632
* allele u-net precision:  0.8737403675163011
* allele u-net recall:  0.7774261603375527

### 50 mixtures:
* allele u-net f1:  0.8113398582517718
* allele u-net precision:  0.8672438672438673
* allele u-net recall:  0.7622067216233355

### 143 mixtures:
* allele u-net f1:  0.8044332375535097
* allele u-net precision:  0.8504649721016739
* allele u-net recall:  0.7631286159323543